### DISCUSSION TITLE
Nextjs Setup with Pino

### DISCUSSION THREAD

**Discussion by besteman** [vercel/next.js#33684] [Help]
I am trying to get Nextjs setup with Pino since it is what is [recommended](https://nextjs.org/docs/going-to-production#logging)

Trying to get this setup so that I can use Pino with Datadog for logging.

Can anyone give me an example, how I would get Pino setup with Nextjs? If it is possible as well maybe Datadog APM (dd-trace) and logging?

**Comment by EmilioAiolfi** [comment]
@besteman maybe this can help you - https://jake.tl/notes/2021-04-04-nextjs-preload-hack

This discussion too https://github.com/vercel/next.js/discussions/16600#discussioncomment-1397255
